### ISSUE TITLE
Multiple type discriminators in JSON silently passes.

### ISSUE THREAD

**Issue by adesam146** [manuc66/JsonSubTypes#100]
Hi, using the following example
```csharp
[JsonConverter(typeof(JsonSubtypes))]
[JsonSubtypes.KnownSubTypeWithProperty(typeof(Employee), "JobTitle")]
[JsonSubtypes.KnownSubTypeWithProperty(typeof(Artist), "Skill")]
public abstract class Person
{
    public string Name { get; set; }
}

public class Employee : Person
{
    public string JobTitle { get; set; }
}

public class Artist : Person
{
    public string Skill { get; set; }
}
```
It appears that if multiple type discriminators are present in the JSON object i.e.
```json
{
  "Name": "Foo",
  "Skill": "A",
  "JobTitle": "B"
}
```
The subtype that is chosen is the subtype registered first in the attribute or via the builder. In this case, that would be the `Employee` type. However, it would be preferable if either an error message was thrown in such a case or the deserialization falls back to the `FallBackSubType`

I believe the function to look at is here https://github.com/manuc66/JsonSubTypes/blob/15a96d3dadf212faaa29a068a5d3d2b0c850ec74/JsonSubTypes/JsonSubtypes.cs#L280-L298
, and the problem could perhaps be addressed by changing `FirstOrDefault` to `SingleOrDefault` in the last line (if the `FallBackSubType` approach is preferred to throwing an exception).

**Comment by manuc66** [comment]
Hi,

First `KnownSubTypeWithProperty` should be chosen to avoid such situation, but if the json comes from untrusted source maybe the most secure is to throw an exception.

A bugfix release can be envisioned to fix this

**Comment by mikeofireland** [comment]
If you create a generic Microsoft API project and create a post route with a [FromBody] Input parameter.  Add further properties outside the scope of the defined object type does not throw an exception by default.

Though I agree this should be up to the developers discretion if it should or shouldn't throw.  By default I would think the current functionality is correct to pick first one that matches.  Though it would be nice if you specifically set the order by which its chosen.

**Comment by manuc66** [comment]
Published with version 1.9.0
